Branch-TestBed/Branch-SDK-Tests/BranchLoggerTests.m

```diff
@@ -11,99 +11,251 @@
-        XCTAssertEqualObjects(message, expectedMessage, "Logged message does not match expected message.");
-        XCTAssertEqual(logLevel, expectedLevel, "Logged level does not match expected level.");
-        XCTAssertNil(error, "Error should be nil.");
-        [expectation fulfill];
+        count = count + 1;
     };
     
+    // one call
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    
+    // disable, second call is ignored
+    logger.loggingEnabled = NO;
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+}
+
+- (void)testLogLevel_DebugByDefault {
+    BranchLogger *logger = [BranchLogger new];
     logger.loggingEnabled = YES;
-    logger.logLevelThreshold = BranchLogLevelInfo;
-    [logger logInfo:@"Test message"];
+    
+    __block int count = 0;
+    logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+        count = count + 1;
+    };
+    
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    [logger logWarning:@"msg" error:nil];
+    XCTAssertTrue(count == 2);
+    [logger logDebug:@"msg" error:nil];
+    XCTAssertTrue(count == 3);
+    
+    // this should be ignored
+    [logger logVerbose:@"msg" error:nil];
+    XCTAssertTrue(count == 3);
+}
 
-    [self waitForExpectationsWithTimeout:1 handler:nil];
+- (void)testLogLevel_Error {
+    BranchLogger *logger = [BranchLogger new];
+    logger.loggingEnabled = YES;
+    logger.logLevelThreshold = BranchLogLevelError;
+    
+    
+    __block int count = 0;
+    logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+        count = count + 1;
+    };
+    
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    
+    // this should be ignored
+    [logger logWarning:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    [logger logDebug:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    [logger logVerbose:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
 }
 
-- (void)testLogLevelSpecificityFiltersLowerLevels {
+- (void)testLogLevel_Warning {
     BranchLogger *logger = [BranchLogger new];
     logger.loggingEnabled = YES;
     logger.logLevelThreshold = BranchLogLevelWarning;
     
-    XCTestExpectation *verboseExpectation = [self expectationWithDescription:@"Verbose log callback"];
-    verboseExpectation.inverted = YES;
-    XCTestExpectation *errorExpectation = [self expectationWithDescription:@"Error log callback"];
     
-    __block NSUInteger callbackCount = 0;
+    __block int count = 0;
+    logger.logCallback = ^(NSString * _Nonnull message, BranchLogLevel logLevel, NSError * _Nullable error) {
+        count = count + 1;
+    };
+    
+    [logger logError:@"msg" error:nil];
+    XCTAssertTrue(count == 1);
+    [logger logWarning:@"msg" error:nil];
+    XCTAssertTrue(count == 2);
+    
+    // this should be ignored
```

Comment:
Can this comment be clarified? Ignored by who, can/should this be removed later?

Reply:
Yep